sdk/inc/azure/core/az_context.h

```diff
@@ -97,12 +96,12 @@ az_context_create_with_value(az_context const* parent, void const* key, void con
 /**
  * @brief Cancels the specified #az_context node; this cancels all the child nodes as well.
  *
- * @param[in,out] ref_context __[nullable]__ A pointer to the #az_context node to be canceled;
- * passing `NULL` cancels the #az_context_application.
+ * @param[in,out] ref_context A pointer to the #az_context node to be canceled.
  */
 AZ_INLINE void az_context_cancel(az_context* ref_context)
 {
-  ref_context = ((ref_context != NULL) ? ref_context : &az_context_application);
+  _az_PRECONDITION_NOT_NULL(ref_context);
```

Comment:
Maybe we should add a precondition that this  context hasn't already been cancelled.

```suggestion
  _az_PRECONDITION_NOT_NULL(ref_context);
  _az_PRECONDITION(ref_context->_internal.expiration != 0);
```
The caller should not try to cancel the same context multiple times, as that hints to a bug in their code.

Reply:
Canceling a context multiple times is valid